### PR TITLE
Include cohere-768 wikipedia embedings dataset generator

### PR DESCRIPTION
**Pull request by filipecosta90**
This dataset is based on Cohere's dataset of [The Embedding Archives: Millions of Wikipedia Article Embeddings in English language](https://txt.cohere.com/embedding-archives-wikipedia/) 

By using the `create_cohere_ds.py` we can specify the train an test size, up to 35M vectors. By default it produces a 1M train size and a 10K test size.

### CODE REVIEW

File: create_cohere_ds.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+
+def train_test_split(X, test_size=10000, dimension=None):
+    import sklearn.model_selection
+
+    if dimension is None:
+        dimension = X.shape[1]
+    print("Splitting %d*%d into train/test" % (X.shape[0], dimension))
+    return sklearn.model_selection.train_test_split(
+        X, test_size=test_size, random_state=1
```
Comment (DvirDukhan) [inline create_cohere_ds.py:62]: Will it stay consistent across all invocations?

File: create_cohere_ds.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    X_train, X_test = train_test_split(X, test_size)
+
+    human_size = human_format(train_size)
+    write_output(
+        train=np.array(X_train),
+        test=np.array(X_test),
+        fn=f"cohere-{dim}-{human_size}-{distance}.hdf5",
+        distance=distance,
+        point_type="float",
+        count=100,
```
Comment (DvirDukhan) [inline create_cohere_ds.py:100]: count is 100?

File: create_cohere_ds.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+@click.command()
+@click.option("--train_size", default=1000000, help="Train size.")
+@click.option("--test_size", default=10000, help="Test size.")
+@click.option("--distance", default="angular", help="distance metric.")
+def create_ds(train_size, test_size, distance):
+    dim = 768
+    total_vecs = train_size + test_size
+    X = np.zeros((total_vecs, dim), dtype=np.float32)
+    from datasets import load_dataset
```
Comment (DvirDukhan) [inline create_cohere_ds.py]: move all imports to the top of the file please

File: create_cohere_ds.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    neighbors = f.create_dataset("neighbors", (len(test), count), dtype="i")
+    distances = f.create_dataset("distances", (len(test), count), dtype="f")
+    bf = BruteForceBLAS(distance, precision=train.dtype)
+
+    bf.fit(train)
+    calc(bf, test, neighbors, distances, count)
+    f.close()
+
+
+def train_test_split(X, test_size=10000, dimension=None):
```
Comment (DvirDukhan) [inline create_cohere_ds.py:55]: I think you drop the content of this function at `create_ds`. There is no actual need for wrapping `train_test_split` with another function

File: create_cohere_ds.py (hunk 1 above)
Comment (filipecosta90) [reply create_cohere_ds.py:62]: according to docs yes: 
> random_state : int, RandomState instance or None, default=None
    Controls the shuffling applied to the data before applying the split. Pass an int for reproducible output across multiple function calls.
    See Glossary <random_state>.

File: create_cohere_ds.py (hunk 3 above)
Comment (filipecosta90) [reply create_cohere_ds.py]: resolved at latest commit

File: create_cohere_ds.py (hunk 4 above)
Comment (filipecosta90) [reply create_cohere_ds.py:55]: addressed in the latest commit

**Review by GuyAv46 — APPROVED** [review]
